File: golang/11_container_with_most_water.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func maxArea(height []int) int {
	area, l, r := 0, 0, len(height)-1
	for l < r {
		temp := 0
		if height[l] < height[r] {
			temp = height[l] * (r - l)
		} else {
			temp = height[r] * (r - l)
		}

		if temp > area {
			area = temp
		}

		if height[l] < height[r] {
			l++
		} else {
			r--
		}
	}
	return area
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	line := strings.Split(scanner.Text(), ",")
	var nums []int
	for i := 0; i < len(line); i++ {
		num, err := strconv.Atoi(line[i])
		if err == nil {
			nums = append(nums, num)
		}
	}
	fmt.Println(maxArea(nums))
}

File: python/11_container_with_most_water.py
from typing import List

class Solution:
    def maxArea(self, height: List[int]) -> int:
        area = 0
        r = len(height) - 1
        l = 0
        while l < r:
            area = max(area, min(height[l], height[r])* (r - l))
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return area

sol = Solution()
inp = [int(x) for x in input().split(",")]
print(sol.maxArea(inp))